Extension/src/LanguageServer/configurations.ts

```diff
@@ -586,8 +586,9 @@ export class CppProperties {
         this.compileCommandFileWatchers = []; //reset it
         let filePaths: Set<string> = new Set<string>();
         this.configurationJson.configurations.forEach(c => {
-            if (c.compileCommands !== undefined && fs.existsSync(c.compileCommands)) {
-                filePaths.add(c.compileCommands);
+            let fileSystemCompileCommandsPath = this.resolvePath(c.compileCommands, os.platform() === "win32");
+            if (c.compileCommands !== undefined && fs.existsSync(fileSystemCompileCommandsPath)) {
```

Comment:
Check `c.compileCommands !== undefined` first before resolving the path?

Reply:
I may have to do the undefined validation first because, resolving the path returns an empty string when path is undefined or ${default}